synapse/util/async.py

```diff
@@ -101,7 +101,7 @@ def remove(r):
             return d
         else:
             success, res = self._result
-            return defer.succeed(res) if success else defer.fail(res)
+            return res if success else defer.fail(res)
```

Comment:
to be clear: I'm not suggesting doing much about it, other than watching for breakage when it lands.